core/Azure.Mcp.Core/src/Extensions/OpenTelemetryExtensions.cs

```diff
@@ -116,11 +101,26 @@ private static void EnableAzureMonitor(this IServiceCollection services)
                     .AddTelemetrySdk();
             });
 
-#if RELEASE
-        otelBuilder.UseAzureMonitorExporter(options =>
+        var appInsightsConnectionStrings = new List<(string Name, string ConnectionString)>();
+
+        var userProvidedAppInsightsConnectionString = Environment.GetEnvironmentVariable("APPLICATIONINSIGHTS_CONNECTION_STRING");
+
+        if (!string.IsNullOrEmpty(userProvidedAppInsightsConnectionString))
         {
-            options.ConnectionString = appInsightsConnectionString;
-        });
+            appInsightsConnectionStrings.Add(("UserProvided", userProvidedAppInsightsConnectionString));
+        }
+
+        var microsoftTelemetry = Environment.GetEnvironmentVariable("AZURE_MCP_COLLECT_TELEMETRY_MICROSOFT");
```

Comment:
* CHANGELOG entry?

Are 2 application insight sources all we will ever export to?  
* If not so, I don't think adding more environment variables is a great experience. (APPLICATION_INSIGHTS_CONNECTION_STRING_B_C_D)
* I would consider using IConfiguration because we control where the application insight strings come from... but that can come after. #1112 
    * We happen to follow only the logic used in open telemetry exporter but we can control how the connection strings are configured.

This is OK.